backend/inspirehep/files/api/mixin.py

```diff
@@ -0,0 +1,133 @@
+import hashlib
+from io import BytesIO
+
+import magic
+import requests
+import structlog
+from flask import current_app
+from inspire_schemas.builders import LiteratureBuilder
+
+from inspirehep.files.api import current_s3_instance
+from inspirehep.records.errors import DownloadFileError
+
+LOGGER = structlog.getLogger()
+
+
+def requests_retry_session(retries=3):
+    session = requests.Session()
+    adapter = requests.adapters.HTTPAdapter(max_retries=retries)
+    session.mount("http://", adapter)
+    session.mount("https://", adapter)
+    return session
+
+
+class FilesMixin:
+    def update(self, data, *args, **kwargs):
+        data = self.add_files(data)
+        super().update(data, *args, **kwargs)
+
+    def add_files(self, data):
+        if not current_app.config.get("FEATURE_FLAG_ENABLE_FILES", False):
+            LOGGER.info("Files are disabled")
+            return data
+
+        if "deleted" in data and data["deleted"]:
+            LOGGER.info("Record is deleted, skipping files.", uuid=self.id)
+            return data
+
+        documents = data.pop("documents", [])
+        figures = data.pop("figures", [])
+
+        self.pop("documents", None)
```

Comment:
this logic is related to literature only, I'd move it to Literature

Reply:
That would be terrible behavior. Using logic from inside mixin here? This is why you create mixin to be able to just add/remove it to add/remove functionality without worrying about additional changes in your code...

Reply:
I actually agree with Pawel, I wouldn't put any files logic in literature

Reply:
Having a mixin handling just literature related files makes no sense though

Reply:
so in this case there should be a LiteratureFilesMixin (inheriting from FilesMixin). Otherwise there will be same mess we had in next.